syncstorage-spanner/src/pool.rs

```diff
@@ -84,6 +87,30 @@ impl SpannerDbPool {
             self.quota,
         ))
     }
+
+    /// Spawn a task to periodically evict idle connections. Calls wrapper sweeper fn
+    ///  to use pool.retain, retaining objects only if they are shorter in duration than
+    ///  defined max_idle.
+    pub fn spawn_sweeper(&self, interval: Duration) {
+        let Some(max_idle) = self.pool.manager().settings.max_idle else {
+            return;
+        };
+        let pool = self.pool.clone();
+        rt::spawn(async move {
+            loop {
+                sweeper(&pool, Duration::from_secs(max_idle.into()));
```

Comment:
in general, prefer constructing something like this `Duration` outside of the loop -- but it's a small, `Copy` type anyway so it doesn't really matter at all here